charts/sonarqube/templates/_helpers.tpl

```diff
@@ -38,37 +38,68 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 
 {{/*
-Set postgresql.secret
+Determine the k8s secret containing the JDBC credentials
 */}}
-{{- define "postgresql.secret" -}}
-{{- if .Values.postgresql.existingSecret -}} 
-{{- .Values.postgresql.existingSecret -}}
-{{- else if .Values.postgresql.enabled -}}
-{{- template "postgresql.fullname" . -}}
+{{- define "jdbc.secret" -}}
+{{- if .Values.postgresql.enabled -}}
+  {{- if .Values.postgresql.existingSecret -}}
+  {{- .Values.postgresql.existingSecret -}}
+  {{- else -}}
+  {{- template "postgresql.fullname" . -}}
+  {{- end -}}
+{{- else if .Values.jdbcOverwrite.enable -}}
+  {{- if .Values.jdbcOverwrite.jdbcSecretName -}}
+  {{- .Values.jdbcOverwrite.jdbcSecretName -}}
+  {{- else -}}
+  {{- template "sonarqube.fullname" . -}}
+  {{- end -}}
 {{- else -}}
-{{- template "sonarqube.fullname" . -}} 
+  {{- template "sonarqube.fullname" . -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set postgresql.secretKey
+Determine JDBC username
 */}}
-{{- define "postgresql.secretPasswordKey" -}}
-{{- if and .Values.postgresql.existingSecretPasswordKey .Values.postgresql.existingSecret -}}
-{{- .Values.postgresql.existingSecretPasswordKey -}}
+{{- define "jdbc.username" -}}
+{{- if and .Values.postgresql.enabled .Values.postgresql.postgresqlUsername -}}
+  {{- .Values.postgresql.postgresqlUsername | quote -}}
+{{- else if and .Values.jdbcOverwrite.enable .Values.jdbcOverwrite.jdbcUsername -}}
+  {{- .Values.jdbcOverwrite.jdbcUsername | quote -}}
 {{- else -}}
-{{- "postgresql-password" -}}
+  {{- "postgres" -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set postgresql.useInternalSecret
+Determine the k8s secretKey contrining the JDBC password
 */}}
-{{- define "postgresql.useInternalSecret" -}}
-{{- if or .Values.postgresql.enabled .Values.postgresql.existingSecret -}} 
-false
-{{- else -}} 
-true
+{{- define "jdbc.secretPasswordKey" -}}
+{{- if .Values.postgresql.enabled -}}
+  {{- if and .Values.postgresql.existingSecret .Values.postgresql.existingSecretPasswordKey -}}
+  {{- .Values.postgresql.existingSecretPasswordKey -}}
+  {{- else -}}
+  {{- "postgresql-password" -}}
```

Comment:
same as previously. the default should be in the `values.yaml` and not hard coded as a fail thru

Reply:
:man_facepalming: I should have had a close look b4 `d992f9a`, this actually is not possible for this logic unfortunately.
This is due to what this logic determines, as it determines the secret key that should be used to set `SONAR_JDBC_PASSWORD` in both deployment types. It was hard-coded before unless it would have been overwritten by the user via `postgresql.secretPasswordKey` and I cant source it from the `values.yaml` since it doesn't get mentioned there anywhere.

What it looked like b4 this change:

https://github.com/SonarSource/helm-chart-sonarqube/blob/af0ae8e5ea1bc7d635a048c1507556948c1a0fe0/charts/sonarqube/templates/_helpers.tpl#L53-62

The Bitnami Chart creates a secret with this key:

https://github.com/bitnami/charts/blob/e83e5dc8d3ad4fa0fdd5001cc3424e4852cf5bd2/bitnami/postgresql/templates/secrets.yaml#L17